bundles/core/org.eclipse.smarthome.core.thing.test/src/test/groovy/org/eclipse/smarthome/core/thing/i18n/ThingStatusInfoI18nLocalizationServiceOSGiTest.groovy

```diff
@@ -0,0 +1,209 @@
+        thing.getHandler().setThingStatusInfo(new ThingStatusInfo(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "@text/offline.with-two-params   [   \"@text/communication-problems\", ,      \"120\"  ]"))
+
+        def thingStatusInfo = thingStatusInfoI18nLocalizationService.getLocalizedThingStatusInfo(thing, null)
+        assertThat thingStatusInfo, is(new ThingStatusInfo(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Thing is offline because of communication problems. Pls try again after 120 seconds."))
+
+        thingStatusInfo = thingStatusInfoI18nLocalizationService.getLocalizedThingStatusInfo(thing, Locale.GERMAN)
+        assertThat thingStatusInfo, is(new ThingStatusInfo(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Thing ist wegen Kommunikationsprobleme offline. Bitte versuchen Sie es in 120 Sekunden erneut."))
+
+        thingStatusInfo = thingStatusInfoI18nLocalizationService.getLocalizedThingStatusInfo(thing, Locale.ENGLISH)
+        assertThat thingStatusInfo, is(new ThingStatusInfo(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Thing is offline because of communication problems. Pls try again after 120 seconds."))
+    }
+
+    @Test
+    void 'assert that unhandled thing can be handled'() {
+        def info = thingStatusInfoI18nLocalizationService.getLocalizedThingStatusInfo(ThingBuilder.create(new ThingTypeUID("xyz:abc"), "123").build(), null)
+        assertThat info, is(ThingStatusInfoBuilder.create(ThingStatus.UNINITIALIZED, ThingStatusDetail.NONE).build())
+    }
+
+    @Test(expected=IllegalArgumentException)
+    void 'assert that given null thing uid is rejected'() {
+        thingStatusInfoI18nLocalizationService.getLocalizedThingStatusInfo(null, null)
+    }
+
+    @Before
+    void setup() {
+        def localeProvider = getService(LocaleProvider)
+        assertThat localeProvider, is(notNullValue())
+        defaultLocale = localeProvider.getLocale()
+
+        setDefaultLocale(Locale.ENGLISH)
+
+        registerVolatileStorageService()
+
+        def simpleThingHandlerFactory = new SimpleThingHandlerFactory()
+        simpleThingHandlerFactory.activate([getBundleContext: { bundleContext }] as ComponentContext)
+        registerService(simpleThingHandlerFactory, ThingHandlerFactory.class.name)
+
+        thing = ThingBuilder.create(new ThingTypeUID("aaa:bbb"), "ccc").build()
+
+        managedThingProvider = getService(ManagedThingProvider)
+        assertThat managedThingProvider, is(notNullValue())
+
+        managedThingProvider.add(thing)
+
+        waitForAssert({
+            assertThat thing.getStatus(), is(ThingStatus.ONLINE)
+        })
+
+        thingStatusInfoI18nLocalizationService = getService(ThingStatusInfoI18nLocalizationService)
+        assertThat thingStatusInfoI18nLocalizationService, is(notNullValue())
+    }
+
+    @After
+    void tearDown() {
+        managedThingProvider.remove(thing.getUID())
+        unregisterMocks()
```

Comment:
Isn't `unregisterMocks()` already be called by the test framework caused by the `@After` the method is using?